Automattic/wp-calypso
# Signup: Abstract social component

This PR abstracts the social part of Signup and Login
<img width="440" alt="image" src="https://github.com/Automattic/wp-calypso/assets/52076348/6c0edfcb-b8f7-447f-8c0d-ba86e797a125">


It creates a shared component in `/blocks/authorization/social`.
This is propedeutic to pdDR7T-10d-p2, as we will try to use the same Social component for both Login and Signup.
Following work will be making Login use this new component!

## Testing
1. Live Link
2. Visit `wordpress.com/log-in` and live link version. Test desktop and mobile.
3. Visit `wordpress.com/start/onboarding-pm` and live link version. Test desktop and mobile.
4. Visit `wordpress.com/start` and live link version. Test desktop and mobile.
5. Visit `wordpress.com/start/videopress-account/user` and live link version. Test desktop and mobile.
6. **P2**: Visit https://wordpress.com/log-in?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fstart%2Fp2%2Fp2-confirm-email&from=p2&signup_url=%2Fstart%2Fp2%2Fuser and live link version. Test desktop and mobile.
7. **Jetpack**: Visit http://calypso.localhost:3000/log-in/jetpack?site=https%3A%2F%2Ffalse-shrike.jurassic.ninja and live link version. Test desktop and mobile.
8. **Crowdsignal**: Visit http://calypso.localhost:3000/log-in?client_id=978&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D51d5e1b0d7358dee3de65c45f4b4dc37a3d6e37dab8d6047545f636c2659d104%26redirect_uri%3Dhttps%253A%252F%252Fapp.crowdsignal.com%252Fconnect%253Fsource%253Dlogin%2526action%253Drequest_access_token%26wpcom_connect%3D1 and live link version. Test desktop and mobile.
9. **Jetpack Cloud**: http://calypso.localhost:3000/log-in?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252F%26scope%3Dglobal%26blog_id%3D0
